src/pyrokinetics/kinetics/imas.py

```diff
@@ -0,0 +1,179 @@
+from pathlib import Path
+from textwrap import dedent
+
+import h5py
+import numpy as np
+from periodictable import elements
+
+from ..constants import deuterium_mass, electron_mass
+from ..equilibrium import Equilibrium
+from ..file_utils import FileReader
+from ..species import Species
+from ..typing import PathLike
+from ..units import UnitSpline
+from ..units import ureg as units
+from .kinetics import Kinetics
+
+
+class KineticsReaderIMAS(FileReader, file_type="IMAS", reads=Kinetics):
+    def read_from_file(
+        self,
+        filename: PathLike,
+        time_index: int = -1,
+        time: float = None,
+        eq: Equilibrium = None,
+    ) -> Kinetics:
+        """
+        Reads in IMAS profiles NetCDF file
+        """
+
+        if eq is None:
+            raise ValueError(
+                dedent(
+                    f"""\
+                    {self.__class__.__name__} must be provided with an Equilibrium object via
+                    the keyword argument 'eq'. Please load an Equilibrium.
+                    """
+                )
+            )
+
+        if time_index is not None and time is not None:
+            raise RuntimeError("Cannot set both 'time' and 'time_index'")
+
+        with h5py.File(filename, "r") as raw_file:
+
+            data = raw_file["core_profiles"]
```

Comment:
```suggestion
        with h5py.File(filename, "r") as raw_file:
            data = raw_file["core_profiles"]
```

And un-indent after this?